manipmongo/sharder.go

```diff
@@ -22,17 +23,23 @@ type Sharder interface {
 
 	// Shard will be call when the shard key needs to be set to
 	// the given elemental.Identifiable.
-	Shard(elemental.Identifiable)
+	Shard(manipulate.TransactionalManipulator, manipulate.Context, elemental.Identifiable) error
+
+	// OnShardedWrite will be called after a successful sharded write
+	// If it returns an error, this error will be returned to the caller
+	// of the manipulate Operation, but the object that has been
+	// created will still be created in database.
+	OnShardedWrite(manipulate.TransactionalManipulator, manipulate.Context, elemental.Operation, elemental.Identifiable) error
```

Comment:
That is confusing for me (as an API user). What does it really mean? Shall I retry? Shall I delete? How do I recover if I see this error?

Reply:
It's more like a hook. I'm still unsure if I should have a Hooker (just for the sake of the name :)) or leave it in the sharder. Errors are returned to manipulator caller